rebalancer.py

```diff
@@ -169,45 +173,55 @@ def auto_enable():
         filter_day = datetime.now() - timedelta(days=apdays)
         forwards = Forwards.objects.filter(forward_date__gte=filter_day)
         for channel in channels:
-            outbound_percent = int(round(channel.outbound_percent/10, 0))
-            inbound_percent = int(round(channel.inbound_percent/10, 0))
-            routed_in_apday = forwards.filter(chan_id_in=channel.chan_id).count()
-            routed_out_apday = forwards.filter(chan_id_out=channel.chan_id).count()
-            iapD = 0 if routed_in_apday == 0 else int(forwards.filter(chan_id_in=channel.chan_id).aggregate(Sum('amt_in_msat'))['amt_in_msat__sum']/10000000)/100
-            oapD = 0 if routed_out_apday == 0 else int(forwards.filter(chan_id_out=channel.chan_id).aggregate(Sum('amt_out_msat'))['amt_out_msat__sum']/10000000)/100
-            if oapD > (iapD*1.10) and outbound_percent > 75:
-                #print('Case 1: Pass')
-                pass
-            elif oapD > (iapD*1.10) and inbound_percent > 75 and channel.auto_rebalance == False:
-                #print('Case 2: Enable AR - o7D > i7D AND Inbound Liq > 75%')
-                channel.auto_rebalance = True
-                channel.save()
-                Autopilot(chan_id=channel.chan_id, peer_alias=channel.alias, setting='Enabled', old_value=0, new_value=1).save()
-            elif oapD < (iapD*1.10) and outbound_percent > 75 and channel.auto_rebalance == True:
-                #print('Case 3: Disable AR - o7D < i7D AND Outbound Liq > 75%')
-                channel.auto_rebalance = False
-                channel.save()
-                Autopilot(chan_id=channel.chan_id, peer_alias=channel.alias, setting='Enabled', old_value=1, new_value=0).save()
-            elif oapD < (iapD*1.10) and inbound_percent > 75:
-                #print('Case 4: Pass')
-                pass
-            else:
-                #print('Case 5: Pass')
-                pass
+            outbound_percent = int(round(channel['outbound_percent']/10, 0))
+            inbound_percent = int(round(channel['inbound_percent']/10, 0))
+            chan_list = lookup_channels.filter(remote_pubkey=channel['remote_pubkey']).values('chan_id')
+            routed_in_apday = forwards.filter(chan_id_in__in=chan_list).count()
+            routed_out_apday = forwards.filter(chan_id_out__in=chan_list).count()
+            iapD = 0 if routed_in_apday == 0 else int(forwards.filter(chan_id_in__in=chan_list).aggregate(Sum('amt_in_msat'))['amt_in_msat__sum']/10000000)/100
+            oapD = 0 if routed_out_apday == 0 else int(forwards.filter(chan_id_out__in=chan_list).aggregate(Sum('amt_out_msat'))['amt_out_msat__sum']/10000000)/100
+
+            for peer_channel in lookup_channels.filter(chan_id__in=chan_list):
```

Comment:
I'm not sure if `lookup_channels` here includes all the individual channels because of the aggregation it should be doing in it's query.

It might be better to move the loop here down into the categories that actually make updates (that way we only loop when we have to write logs).

Reply:
I have validated that `lookup_channels` does contain all the individual channels as instantiated in line 178. The aggregation and filters create a new object based on the query set.  This approach also prevents multiple database access. 
Can be confirmed in the logs by uncommenting the next print statement.